### PR TITLE
Add missing execution resources to fetched traces

### PR DESCRIPTION
**Pull request by kirugan**

### CODE REVIEW

File: rpc/trace.go [hunk 1]
```diff
@@ -215,6 +215,20 @@ func (h *Handler) traceBlockTransactions(ctx context.Context, block *core.Block,
...
+			if err != nil {
+				return nil, httpHeader, err
+			}
+
+			if !v0_6Response {
+				// add execution resources on root level
+				for index, trace := range result {
+					executionResources := trace.TraceRoot.TotalExecutionResources()
+					// fgw doesn't provide this data in traces endpoint
+					executionResources.DataAvailability = &vm.DataAvailability{}
```
Comment (rianhughes) [inline rpc/trace.go:227]: I guess DataAvailability shouldn't be zero. I think we can fetch it from the Receipts

**Comment by rianhughes** [comment]
closes https://github.com/NethermindEth/juno/issues/2218

File: rpc/trace.go (hunk 1 above)
Comment (kirugan) [reply rpc/trace.go]: As you suggested I've added data availability from receipts and it works for some blocks (for very old ones like sepolia #7 it doesn't work)

**Comment by rianhughes** [comment]
LGTM, I'll approve when the tests are fixed

**Review by IronGauntlets — APPROVED** [review]

**Review by rianhughes — APPROVED** [review]

**Pull request merged**